Emby.Server.Implementations/Library/Validators/CollectionPostScanTask.cs

```diff
@@ -0,0 +1,152 @@
+                {
+                    if (collectionNameMoviesMap.TryGetValue(movie.CollectionName, out var movieList))
+                    {
+                        if (!movieList.Any(m => m.Id == movie.Id))
+                        {
+                            movieList.Add(movie);
+                        }
+                    }
+                    else
+                    {
+                        collectionNameMoviesMap[movie.CollectionName] = new List<Movie> { movie };
+                    }
+
+                }
+
+                numComplete++;
+                double percent = numComplete;
+                percent /= count * 2;
+                percent *= 100;
+
+                progress.Report(percent);
+            }
+
+            foreach (var (collectionName, movieList) in collectionNameMoviesMap)
+            {
+                try
+                {
+                    var boxSet = boxSets.FirstOrDefault(b => b?.Name == collectionName) as BoxSet;
+                    if (boxSet == null)
+                    {
+                        // won't automatically create collection if only one movie in it
+                        if (movieList.Count >= 2)
+                        {
+                            var movieIds = FliterMoviesByOption(movieList);
+                            if (movieIds.Count >= 2) {
+                                // at least 2 movies have AutoCollection option enable
+                                boxSet = await _collectionManager.CreateCollectionAsync(new CollectionCreationOptions
+                                {
+                                    Name = collectionName,
+                                    IsLocked = true
+                                });
+
+                                await _collectionManager.AddToCollectionAsync(boxSet.Id, movieIds);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        var movieIds = FliterMoviesByOption(movieList);
+                        await _collectionManager.AddToCollectionAsync(boxSet.Id, movieIds);
+                    }
+
+                    numComplete++;
+                    double percent = numComplete;
+                    percent /= count * 2;
+                    percent *= 100;
+
+                    progress.Report(percent);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error refreshing {CollectionName} with {@MovieIds}", collectionName, movieList);
+                }
+            }
+
+            progress.Report(100);
+        }
+
+        private List<Guid> FliterMoviesByOption(List<Movie> movieList) {
+            List<Guid> movieIds = new List<Guid>();
+            foreach (var movie in movieList)
+            {
+                if (_libraryManager.GetLibraryOptions(movie).AutoCollection)
```

Comment:
I worry that this is inefficient. It is more likely that a user has a handful of movie libraries _at most_, but they could contain thousands of movies. That is a lot of unnecessary calls.

Maybe the whole thing should be flipped. If you look at this task https://github.com/jellyfin/jellyfin/blob/f5ca9cbc3bcffc13830669d5790525c6c11f9136/MediaBrowser.Providers/MediaInfo/SubtitleScheduledTask.cs#L73, you can see how it fetches items for specific libraries. We could have it do

1. Grab all libraries
2. Find libraries for which `AutoCollection` is true (note: I think the name should be changed to something like `AutomaticallyCreateCollections`)
3. Find all movies in the libraries (with AutoCollection=true) https://github.com/jellyfin/jellyfin/blob/f5ca9cbc3bcffc13830669d5790525c6c11f9136/MediaBrowser.Providers/MediaInfo/SubtitleScheduledTask.cs#L99-L108
4. Create the mapping like you do now
5. Create collections

Reply:
Great! I want to do this before, but I dont' know how to query movies in specfic library. Thank you.

As for library option, I choose `AutoCollection` beacause it not just create. If a collection already exits, it also will automatic add movie to collection.